pages/api/editclass.js

```diff
@@ -0,0 +1,41 @@
+import prisma from '../../prisma/prisma';
+import { unstable_getServerSession } from 'next-auth';
+import { authOptions } from './auth/[...nextauth]';
+
+export default async function handle(req, res) {
+  // unstable_getServerSession is recommended here: https://next-auth.js.org/configuration/nextjs
+  const session = await unstable_getServerSession(req, res, authOptions);
+  const data = req.body;
+
+  if (!req.method == 'PUT') {
+    res.status(405).end();
+  }
+
+  if (!session) {
```

Comment:
should we only check for a session? or should we check that whoever is making this req to the API is a teacher